### PR TITLE
Fix response for no CDAWEB results

### PR DESCRIPTION
**Pull request by dstansby**
Fixes https://github.com/sunpy/sunpy/issues/6177

### CODE REVIEW

File: sunpy/net/cdaweb/cdaweb.py [hunk 1]
```diff
@@ -94,17 +94,21 @@ def _do_search(self, query):
...
-        etimes = [f['EndTime'] for f in response['FileDescription']]
-        urls = [f['Name'] for f in response['FileDescription']]
-
-        return astropy.table.QTable(
-            {'Dataset': [query['dataset']] * len(stimes),
-             'Start time': Time.strptime(stimes, '%Y-%m-%dT%H:%M:%S.%fZ').iso,
-             'End time': Time.strptime(etimes, '%Y-%m-%dT%H:%M:%S.%fZ').iso,
-             'URL': urls})
+            # No results
+            return astropy.table.QTable(
```
Comment (Cadair) [inline sunpy/net/cdaweb/cdaweb.py:98]: This shouldn't be returning a QTable at all it should be returning a QueryResponseTable (which is a subclass of QTable).

Comment (dstansby) [reply sunpy/net/cdaweb/cdaweb.py:98]: This is an internal helper method, isn't it fine that this returns whatever as long as `search` returns the `QueryResponseTable` (which it does above)?

**Review by nabobalis — APPROVED** [review]

**Review by wtbarnes — APPROVED** [review]

**Comment by dstansby** [comment]
Woops, I've fixed the changelog name. Accidentally closed and re-opened this PR, will merge when CI passes....

File: sunpy/net/cdaweb/cdaweb.py (hunk 1 above)
Comment (nabobalis) [reply sunpy/net/cdaweb/cdaweb.py:98]: I think if we are worried about this, we can follow up in a new PR but it works and I don't see the issue with it being a Qtable.

**Pull request merged**